app/Community/Concerns/ActsAsCommunityMember.php

```diff
@@ -28,6 +29,30 @@ public static function bootActsAsCommunityMember(): void
     {
     }
 
+    public function getVisibleRoleAttribute(): ?SpatieRole
+    {
+        // Load the user's displayable roles.
+        if ($this->relationLoaded('displayableRoles')) {
+            $displayableRoles = $this->displayableRoles;
+        } else {
+            $displayableRoles = $this->displayableRoles()->orderBy('display')->get();
+        }
+
+        // If user has an explicitly set visible_role_id, we'll try to show it.
+        // However, we need to verify it's still a valid displayable role for the
+        // user (it's possible they lost the role at some point).
+        if ($this->visible_role_id !== null) {
+            $explicitRole = $displayableRoles->find($this->visible_role_id);
+            if ($explicitRole) {
+                return $explicitRole;
+            }
+        }
+
+        // Otherwise, fall back to highest ordered displayable role.
+        // For most users, this will return null.
+        return $displayableRoles->first();
+    }
```

Comment:
Not sure exactly what's going on here, but it looks like this query gets run 6 times (for like 60% of the page time).
![image](https://github.com/user-attachments/assets/6135dab3-96cf-45c9-869f-3084fa2ca166)

Reply:
Fixed in latest.